openmdao/core/vec_wrapper.py

```diff
@@ -278,7 +278,7 @@ def _get_scalar_units_complex(self):
 
     def _set_arr(self, value):
         """Set an array value."""
-        self.val[:] = value.flat
+        self.val[:] = value.real.flat
```

Comment:
ok. I removed it. @naylor-b was right that it was unnecessary. The message comes from the component itself, where the user is being a bit lazy and not cleaning up their complex stuff correctly